### ISSUE TITLE
[Feature Request] Consider tighter MSI validation

### ISSUE THREAD

**Issue by DaveTryon** [microsoft/accessibility-insights-windows#731]
VersionSwitcher currently checks only that the new MSI has a valid and non-revoked digital signature, and disregards _who_ signed the MSI. We might want to tighten this check to ensure that VersionSwitcher only installs _our_ MSI's. Here is a non-exhaustive list of options:

1. Validate the canonical name of the MSI's signing certificate against a collection of names built into the VersionSwitcher. Refuse to install if validation fails.
2. Validate the thumbprint of the MSI's signing certificate against a collection of thumbprints built into the VersionSwitcher. Refuse to install if validation fails.
3. Validate the canonical name or thumbprint of the MSI's signing certificate against data that gets added to the web-hosted metadata file. Pass the validation as a command line parameter into VersionSwitcher, Refuse to install if validation fails. 
4. Validate a hash of the MSI file against data that gets added to the web-hosted metadata file. Pass the validation as a command line parameter into VersionSwithcer. Refuse to install if validation fails.

**Comment by sauravbhattacharya001** [comment]
There is some code in trust verifier: IsFileTrusted(string filePath). Can a similar be used in VersionSwitcher to help with this?

**Comment by sauravbhattacharya001** [comment]
where can I find the canonical names and thumbprints?

**Comment by DaveTryon** [comment]
@sauravbhattacharya001, the various options were out there to reflect the fact that there was some uncertainty about our long-term signing strategy. That uncertainty has since been worked out, and our plan for the foreseeable future is to use a certificate with the canonical name of **Microsoft Corporation** (that's the certificate used to sign the MSI files published at  https://github.com/Microsoft/accessibility-insights-windows/releases). With that decision in place, we'd probably opt for the first (and simplest) option listed above, structuring the code to use any match from a hardcoded list of canonical names, then only putting one item in the list--that makes it simple to support different canonical names should future decisions require it. We prefer the canonical name to a thumbprint because it represents less churn as certificates are renewed.

This approach won't require the modification of the web-hosted metadata file, but just to answer your question, the code in the [GitHubAutoUpdate project](https://github.com/microsoft/accessibility-insights-windows/tree/master/src/AccessibilityInsights.Extensions.GitHubAutoUpdate) selects a metadata file based on the release channel, either [Canary](https://github.com/microsoft/accessibility-insights-windows/blob/Canary/Channels/Canary/release_info.json), [Insider](https://github.com/microsoft/accessibility-insights-windows/blob/Control/Channels/Insider/release_info.json), or [Production](https://github.com/microsoft/accessibility-insights-windows/blob/Control/Channels/Production/release_info.json). The metadata files are stored in branches that integrate with our release pipelines

**Comment by DaveTryon** [comment]
Update: After further conversation, the scenario we really wanted to avoid was this:
- Malicious actor launches VersionSwitcher with a crafty command line
- VersionSwitcher prompts for elevation and is displayed as owned by Microsoft
- User grants elevation
- VersionSwitcher installs a malicious MSI (which won't prompt since we're already elevated)
- User has been compromised

The best way we found to avoid that scenario was to bake the decision of "do we trust this MSI" into VersionSwitcher, since anything that uses a command line parameter would be vulnerable to a malicious actor.

After talking with the team that manages the signing certificates, they suggested that we validate against the issuer of the signing cert, as opposed to its subject. To this end, we added a list of organizations whose issued certs we will trust. At the moment, "Microsoft Corporation" is the only organization in that list, but we have the option to change it should the need arise in the future.

I'll close this after the change deploys to Canary and I can validate there.
